proto file complete

RayMeng1998 opened the pull request:
completed proto file. wait for feedback

RayMeng1998 committed:
add api resource

RayMeng1998 committed:
complete proto file

yfei1 commented on .idea/modules.xml:
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Please add the .idea/ dir and BIN files to .gitignore

yfei1 replied:
Also please update your commit message to be more meaningful.

yfei1 commented on examples/helloworld/helloworld/api_resources.proto:
```diff
@@ -0,0 +1,102 @@
+
+
+message Business {
+  string id = 1;
```
How about opening hours? Please refer to https://developers.google.com/protocol-buffers/docs/proto3 and implement it.

yfei1 commented on examples/helloworld/helloworld/api_resources.proto:
```diff
@@ -0,0 +1,102 @@
+  string user_id = 1;
+}
+
+message DeleteUserReply {
```
Generally speaking, we often use google.protobuf.Empty as the DeleteReply. See https://google.aip.dev/135 for more details.

yfei1 commented on examples/helloworld/helloworld/api_resources.proto:
```diff
@@ -0,0 +1,102 @@
+}
+
+message GetUserReply {
+  string user_id = 1;
```
`public string GetUser(string) {} ` looks weird, can you give it some more thoughts?

yfei1 commented on examples/helloworld/helloworld/api_resources.proto:
```diff
@@ -0,0 +1,102 @@
+}
+
+message DeleteUserRequest {
+  string user_id = 1;
```
See the comment below.

yfei1 commented on examples/helloworld/helloworld/api_resources.proto:
```diff
@@ -0,0 +1,102 @@
+}
+
+message GetBusinessReply {
+  string business_id = 1;
```
Same.

yfei1 commented on examples/helloworld/helloworld/api_resources.proto:
```diff
@@ -0,0 +1,102 @@
+}
+
+message DeleteBusinessReply {
+  string business_id = 1;
```
Same.

yfei1 commented on examples/helloworld/helloworld/review.proto:
```diff
@@ -0,0 +1,30 @@
+  repeated Review child_reviews = 8;
+}
+
+message CreateReviewRequest {
```
Where is this being used?

yfei1 commented on examples/helloworld/helloworld/review.proto:
```diff
@@ -0,0 +1,30 @@
+  Review review = 1;
+}
+
+message UpdateReviewRequest {
```
Here and below, same.

RayMeng1998 committed:
fixed the proto file as requested

Fixed the delete reply and create/update review.

Not sure how to add .idea to .gitgore

RayMeng1998 committed:
added operating hours; fixed delete reply

Added operating hours by specifying starting minute and closing minute of a day.
Added open days using an enum type.
Had the delete interface returns google.protobuf.Empty directly

RayMeng1998 committed:
integrated metadata into api_resources

RayMeng1998 committed:
Update api_resources.pb.go
